src/main/java/bio/terra/workspace/service/datareference/DataReferenceService.java

```diff
@@ -53,7 +53,8 @@ public DataReferenceService(
   public DataReference getDataReference(
       UUID workspaceId, UUID referenceId, AuthenticatedUserRequest userReq) {
 
-    samService.workspaceAuthz(userReq, workspaceId, SamConstants.SAM_WORKSPACE_READ_ACTION);
+    workspaceService.authorizedGetWorkspace(
```

Comment:
For assertions that throw, I like to prefix the function with `validate`, though I don't know if there's a prevailing convention here.

More importantly, this method checks more than workspace read, so maybe `validateWorkspaceOperation()` would be better.

Reply:
I hate naming! I want to also indicate in the name that this method retrieves and returns a `Workspace` object, after validating permissions. I'll try to come up with something better.

Reply:
renamed to `validateWorkspaceAndAction`, how's that? The codebase uses "action" consistently instead of "operation" for the thing in question here.

Reply:
sounds good